pydantic_evals/pydantic_evals/evals.py

```diff
@@ -0,0 +1,302 @@
+from __future__ import annotations as _annotations
+
+import asyncio
+from collections import defaultdict
+from collections.abc import Awaitable
+from contextlib import nullcontext
+from contextvars import ContextVar
+from dataclasses import dataclass, field
+from typing import Any, Callable, Generic
+
+import logfire
+from pydantic_core import to_jsonable_python
+from typing_extensions import TypeVar
+
+from ._utils import get_unwrapped_function_name
+from .assessments.spec import Assessment, AssessmentDetail, AssessmentSpec, BoundAssessmentFunction, ScoringContext
+from .datasets import EvaluationRow
+from .otel.context_in_memory_span_exporter import context_subtree_spans
+from .otel.span_tree import SpanTree
+from .reporting.reports import EvalReport, EvalReportCase, EvalReportCaseAggregate
+
+# while waiting for https://github.com/pydantic/logfire/issues/745
+try:
+    import logfire._internal.stack_info
+except ImportError:
+    pass
+else:
+    from pathlib import Path
+
+    logfire._internal.stack_info.NON_USER_CODE_PREFIXES += (str(Path(__file__).parent.absolute()),)
+
+__all__ = ('Evaluation', 'increment_eval_metric')
```

Comment:
```suggestion
__all__ = ('Evaluation', 'increment_eval_metric', 'set_eval_attribute')
```

Reply:
should there be a convenience function here for recording metrics without incrementing?

Reply:
I used to have one but I couldn't imagine using it